pilot/pkg/networking/plugin/mixer/mixer.go

```diff
@@ -333,3 +333,31 @@ func addStandardNodeAttributes(attr map[string]*mpb.Attributes_AttributeValue, p
 		}
 	}
 }
+
+func serviceConfig(serviceHostname string, dest *model.ServiceInstance, config model.IstioConfigStore, disableCheck, disableReport bool) *mccpb.ServiceConfig {
+	sc := v1.ServiceConfig(serviceHostname, dest, config, disableCheck, disableReport)
+	addDestinationServiceAttributes(sc.MixerAttributes.Attributes, serviceHostname)
+	return sc
+}
+
+func addDestinationServiceAttributes(attrs map[string]*mpb.Attributes_AttributeValue, destinationHostname string) {
+	svcName, svcNamespace := nameAndNamespace(destinationHostname)
+	attrs[v1.AttrDestinationService] = attrStringValue(destinationHostname)
+	attrs["destination.service.host"] = attrStringValue(destinationHostname)
+	attrs["destination.service.uid"] = attrStringValue(fmt.Sprintf("istio://%s/services/%s", svcNamespace, svcName))
+	attrs["destination.service.name"] = attrStringValue(svcName)
+	attrs["destination.service.namespace"] = attrStringValue(svcNamespace)
+}
+
+func nameAndNamespace(serviceHostname string) (name, namespace string) {
+	if !strings.HasSuffix(serviceHostname, "svc.cluster.local") {
```

Comment:
not sure if you care, but there is `DomainSuffix` variable that can be used here in place of `svc.cluster.local` to allow non-default cluster suffixes.

Reply:
I do care. Any ideas on the best way to plumb that through appropriately?

Reply:
I would just use the proxy's domain for the moment. We stopped plumbing this DomainSuffix thign everywhere. Its all FQDNs throughout the code.

Reply:
@rshriram "for the moment" has me a bit nervous -- what is the replacement? proxy's domain also potentially includes the namespace, right?  Is that still OK to use?  Otherwise, I could (read: i was in-process of doing this) add `DomainSuffix` to the `Env`.